pkg/tsdbctl/check.go

```diff
@@ -91,58 +126,172 @@ func (cc *checkCommandeer) check() error {
 	}
 
 	// Get metric data and metadata
-	allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
 	container, tablePath := cc.rootCommandeer.adapter.GetContainer()
-	objPath := fmt.Sprintf("%s/%s", tablePath, cc.objPath)
+
+	var respChans []chan *v3io.Response
+	if cc.metricName == "" {
+		respChan, err := cc.checkByPath(container, tablePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check using object path '%s'", cc.objPath)
+		}
+		respChans = []chan *v3io.Response{respChan}
+		objPath := "/" + path.Join("", tablePath, cc.objPath)
+		allAttrs := append(cc.attrs, "__name", "_name", "_lset", "_maxtime")
+		input := v3io.GetItemInput{Path: objPath, AttributeNames: allAttrs}
+		_, err = container.GetItem(&input, nil, respChan)
+		if err != nil {
+			return errors.Wrap(err, "failed to send GetItem request")
+		}
+	} else {
+		respChans, err = cc.checkByName(container, tablePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to check using metric name (%s) and labels (%v)", cc.metricName, cc.labels)
+		}
+	}
+
+	for _, respChan := range respChans {
+		resp := <-respChan
```

Comment:
I think it could hang on waiting on channel - add timeout

Reply:
That's the responsibility of v3io-go-http. I'm actually adding a timeout API to it right now. See https://github.com/v3io/v3io-go-http/pull/10